src/ui/settings/index.js

```diff
@@ -24,7 +24,14 @@ export default function setupSettings(kbnServer, server, config) {
   }
 
   function getUserProvided() {
-    const { client } = server.plugins.elasticsearch;
+    const elasticsearchPlugin = server.plugins.elasticsearch;
+
+    // If elasticsearch plugin is not ready, do not block
+    if (elasticsearchPlugin.status.state === 'red') {
+      return Promise.resolve({});
```

Comment:
Maybe we should log something, at least at a debug level, about returning early at this point?